numpy/lib/recfunctions.py

```diff
@@ -972,6 +1016,36 @@ def structured_to_unstructured(arr, dtype=None, copy=False, casting='unsafe'):
                                  'itemsize': arr.dtype.itemsize})
     arr = arr.view(flattened_fields)
 
+    # we only allow a few types to be unstructured by manipulating the
+    # strides, because we know it won't work with, for example, np.matrix nor
+    # np.ma.MaskedArray.
+    can_view = type(arr) in (np.ndarray, np.recarray, np.memmap)
+    if (not copy) and can_view and all(dt.base == out_dtype for dt in dts):
+        # all elements have the right dtype already; if they have a common
+        # stride, we can just return a view
+        common_stride = _common_stride(offsets, counts, out_dtype.itemsize)
+        if common_stride is not None:
+            wrap = arr.__array_wrap__
+
+            new_shape = arr.shape + (sum(counts), out_dtype.itemsize)
+            new_strides = arr.strides + (abs(common_stride), 1)
+
+            arr = arr[..., np.newaxis].view(np.uint8)  # view as bytes
+            arr = arr[..., min(offsets):]  # remove the leading unused data
+            arr = np.lib.stride_tricks.as_strided(arr,
+                                                  new_shape,
+                                                  new_strides,
+                                                  subok=True)
+
+            # cast and drop the last dimension again
+            arr = arr.view(out_dtype)[..., 0]
+
+            if common_stride < 0:
+                arr = arr[..., ::-1]  # reverse, if the stride was negative
+            if type(arr) is not type(wrap.__self__):
```

Comment:
There is no need to do this -- by giving `subok=True`, the wrapping is already done in `as_strided` (and off course the various views are already OK). If you want to be sure, I suggest adding a test.

Reply:
I have added a test, that's how I discovered that this is necessary.

It seems the `recarray` turns into a `ndarray` once I do the `view(np.uint8)`. I don't think that's usually a problem, because a `recarray` isn't very useful for a non-structured dtype.
But the old behavior is to keep the subclass anyway (which still happens with `copy=True`) and I don't want to change that without a good reason.

Reply:
Ah, fair enough, another class that is showing its age (and informally deprecated...), though I guess it is logical. It is a pity `as_strided` doesn't take an offsets argument!

But for here, maybe add a note in the code for why we do the wrap?

Reply:
Done